src/charm.py

```diff
@@ -492,6 +534,14 @@ def upgrade_in_progress(self) -> bool:
         """
         return False
 
+    @property
+    def config_server_db(self) -> str:
+        """Fetch current the config server database that this unit is connected to."""
+        if not self.is_integrated_to_config_server():
+            return ""
+
+        return self.model.get_relation(Config.Relations.CLUSTER_RELATIONS_NAME).app.name
```

Comment:
I have the feeling we are duplicating calls here:
* `self.is_integrated_to_config_server()` already get the relation but only checks for its existence
* We then get the relation again to return the value

In terms of design, I'm wondering if the `is_integrated_to_config_server` shouldn't be `get_config_server_integration` which returns a `Relation | None` and we then check for nullity.
This would allow to get-and-store instead of double get (and maybe (? I'm not sure) avoid the case where first check succeeds and second get returns `None` (is it possible?))

Reply:
Hi Neha. I agree with the duplicated calls, not a good look 🙈 .

 But I would prefer not to make a trade off and obscure the naming + return values by modifying `is_integrated_to_config_server`

But still I think your intuition is right here. I updated both to reduce duplicate calls using your fav the walrus operator. Please let me know if you think this is a good comprise

Reply:
It's good yeah!